xpk.py

```diff
@@ -1611,7 +1838,7 @@ def set_jobset_on_cluster(args) -> int:
   """
   command = (
       'kubectl apply --server-side -f'
-      ' https://github.com/kubernetes-sigs/jobset/releases/download/v0.3.2/manifests.yaml'
+      ' https://github.com/kubernetes-sigs/jobset/releases/download/v0.4.0/manifests.yaml'
```

Comment:
I would recommend moving this version bump into a separate PR unless pathways is requiring this somehow

Reply:
This is actually needed for Pathways. This version enabled exclusive-topology at replicatedJobs level, which is helping us scale up cpu nodepools.

Reply:
Makes sense to bump this. The con of bumping it as part of this PR is that we now multiple points of potential breakage from one PR. 

Having it in a separate PR allows for integ tests and for us to check for issues against the version change specifically.